src/event.rs

```diff
@@ -0,0 +1,566 @@
+		/// The channel_id of the channel which has been closed.
+		channel_id: [u8; 32],
+		/// The user_channel_id of the channel which has been closed.
+		user_channel_id: u64,
+	},
+	// TODO: Implement on-chain events when better integrating with BDK wallet sync.
+	//OnChainPaymentSent {
+	//},
+	//OnChainPaymentReceived {
+	//}
+}
+
+// TODO: Figure out serialization more concretely - see issue #30
+impl Readable for LdkLiteEvent {
+	fn read<R: lightning::io::Read>(
+		reader: &mut R,
+	) -> Result<Self, lightning::ln::msgs::DecodeError> {
+		match Readable::read(reader)? {
+			0u8 => {
+				let payment_hash: PaymentHash = Readable::read(reader)?;
+				Ok(Self::PaymentSuccessful { payment_hash })
+			}
+			1u8 => {
+				let payment_hash: PaymentHash = Readable::read(reader)?;
+				Ok(Self::PaymentFailed { payment_hash })
+			}
+			2u8 => {
+				let payment_hash: PaymentHash = Readable::read(reader)?;
+				let amount_msat: u64 = Readable::read(reader)?;
+				Ok(Self::PaymentReceived { payment_hash, amount_msat })
+			}
+			3u8 => {
+				let channel_id: [u8; 32] = Readable::read(reader)?;
+				let user_channel_id: u64 = Readable::read(reader)?;
+				Ok(Self::ChannelReady { channel_id, user_channel_id })
+			}
+			4u8 => {
+				let channel_id: [u8; 32] = Readable::read(reader)?;
+				let user_channel_id: u64 = Readable::read(reader)?;
+				Ok(Self::ChannelClosed { channel_id, user_channel_id })
+			}
+			//5u8 => {
+			// TODO OnChainPaymentSent
+			//}
+			//6u8 => {
+			// TODO OnChainPaymentReceived
+			//}
+			_ => Err(lightning::ln::msgs::DecodeError::InvalidValue),
+		}
+	}
+}
+
+impl Writeable for LdkLiteEvent {
+	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), lightning::io::Error> {
+		match self {
+			Self::PaymentSuccessful { payment_hash } => {
+				0u8.write(writer)?;
+				payment_hash.write(writer)?;
+				Ok(())
+			}
+			Self::PaymentFailed { payment_hash } => {
+				1u8.write(writer)?;
+				payment_hash.write(writer)?;
+				Ok(())
+			}
+			Self::PaymentReceived { payment_hash, amount_msat } => {
+				2u8.write(writer)?;
+				payment_hash.write(writer)?;
+				amount_msat.write(writer)?;
+				Ok(())
+			}
+			Self::ChannelReady { channel_id, user_channel_id } => {
+				3u8.write(writer)?;
+				channel_id.write(writer)?;
+				user_channel_id.write(writer)?;
+				Ok(())
+			}
+			Self::ChannelClosed { channel_id, user_channel_id } => {
+				4u8.write(writer)?;
+				channel_id.write(writer)?;
+				user_channel_id.write(writer)?;
+				Ok(())
+			} //Self::OnChainPaymentSent { .. } => {
+			  //TODO
+			  //}
+			  //Self::OnChainPaymentReceived { .. } => {
+			  //TODO
+			  //}
+		}
+	}
+}
+
+pub(crate) struct LdkLiteEventQueue<K: KVStorePersister> {
+	queue: Mutex<EventQueueSerWrapper>,
+	notifier: Condvar,
+	persister: Arc<K>,
+}
+
+impl<K: KVStorePersister> LdkLiteEventQueue<K> {
+	pub(crate) fn new(persister: Arc<K>) -> Self {
+		let queue: Mutex<EventQueueSerWrapper> = Mutex::new(EventQueueSerWrapper(VecDeque::new()));
```

Comment:
This sort of usage came up when drafting the offers serialization. We ultimately decided to only use serialization wrappers at read/write time.

Reply:
Yes, this was already commented on above (see https://github.com/lightningdevkit/ldk-lite/pull/10#discussion_r980576724). The reason I went with this approach is mainly that it allows the `EventQueue` to not implement `Readable`/`Writeable` itself and hence is able to call `persist` manually with an already unlocked queue.

Let me know if this is acceptable, or if I should rethink this design.

Reply:
Now not holding the wrappers anymore with https://github.com/lightningdevkit/ldk-lite/pull/10/commits/9e3e9db532ef211913fe9f54849aabaa6195fcc5.